pkg/reconciler/subscription/subscription.go

```diff
@@ -189,32 +184,70 @@ func (r *Reconciler) reconcile(ctx context.Context, subscription *v1alpha1.Subsc
 		return err
 	}
 
-	subscriberURI, err := eventingduck.SubscriberSpec(ctx, r.DynamicClientSet, subscription.Namespace, subscription.Spec.Subscriber, r.addressableTracker, trackAddressable)
-	if err != nil {
-		logging.FromContext(ctx).Warn("Failed to resolve Subscriber",
-			zap.Error(err),
-			zap.Any("subscriber", subscription.Spec.Subscriber))
-		r.Recorder.Eventf(subscription, corev1.EventTypeWarning, subscriberResolveFailed, "Failed to resolve spec.subscriber: %v", err)
-		subscription.Status.MarkReferencesNotResolved(subscriberResolveFailed, "Failed to resolve spec.subscriber: %v", err)
-		return err
-	}
-
-	subscription.Status.PhysicalSubscription.SubscriberURI = subscriberURI
-	logging.FromContext(ctx).Debug("Resolved Subscriber", zap.String("subscriberURI", subscriberURI))
+	subscriber := subscription.Spec.Subscriber
+	subscription.Status.PhysicalSubscription.SubscriberURI = nil
+	if !isNilOrEmptySubscriber(subscriber) {
+		// Populate the namespace for the subscriber since it is in the namespace
+		if subscriber.Ref != nil {
+			subscriber.Ref.Namespace = subscription.Namespace
+		}
+		subscriberURIStr, err := r.destinationResolver.URIFromDestination(*subscriber, subscription)
+		if err != nil {
+			logging.FromContext(ctx).Warn("Failed to resolve Subscriber",
+				zap.Error(err),
+				zap.Any("subscriber", subscriber))
+			r.Recorder.Eventf(subscription, corev1.EventTypeWarning, subscriberResolveFailed, "Failed to resolve spec.subscriber: %v", err)
+			subscription.Status.MarkReferencesNotResolved(subscriberResolveFailed, "Failed to resolve spec.subscriber: %v", err)
+			return err
+		}
+		subscriberURI, err := apis.ParseURL(subscriberURIStr)
+		if err != nil {
+			logging.FromContext(ctx).Warn("Failed to parse Subscriber URL",
+				zap.Error(err),
+				zap.Any("subscriber", subscriber))
+			r.Recorder.Eventf(subscription, corev1.EventTypeWarning, subscriberResolveFailed, "Failed to parse URL for spec.subscriber: %v", err)
+			subscription.Status.MarkReferencesNotResolved(subscriberResolveFailed, "Failed to parse URL for spec.subscriber: %v", err)
+			return err
+		}
+		subscription.Status.PhysicalSubscription.SubscriberURI = subscriberURI
+		logging.FromContext(ctx).Debug("Resolved Subscriber", zap.String("subscriberURI", subscriberURIStr))
+	}
+
+	reply := subscription.Spec.Reply
+	subscription.Status.PhysicalSubscription.ReplyURI = nil
+	subscription.Status.ClearDeprecated()
+	if !isNilOrEmptyReply(reply) {
+		// Populate the namespace for the subscriber since it is in the namespace
+		if reply.Channel.Ref != nil {
+			reply.Channel.Ref.Namespace = subscription.Namespace
+		} else {
+			reply.Channel.DeprecatedNamespace = subscription.Namespace
+			// Add a condition warning that the fields are deprecated.
+			subscription.Status.MarkReplyDeprecatedRef(replyFieldsDeprecated, "Using depreated fields when specifying subscription.spec.reply. These will be removed in 0.11")
+		}
```

Comment:
This is implicitly saying that the URI alone shouldn't be used. Is that correct? And if so, is anything validating that?